src/balancerd/src/service.rs

```diff
@@ -39,6 +40,16 @@ pub struct Args {
         requires_all = &["frontegg-jwk", "frontegg-api-token-url", "frontegg-admin-role"],
     )]
     frontegg_resolver_template: Option<String>,
+    /// HTTPS resolver address template. `{}` is replaced with the first subdomain of the HTTPS SNI
+    /// host address to get a DNS address. The first IP that address resolves to is the proxy
+    /// destinaiton.
+    #[clap(long, value_name = "HOST.{}.NAME:PORT")]
+    https_resolver_template: String,
+    /// HTTPS SNI extraction regex. Using the full SNI servername, the first capture group is
+    /// provided as the `{}` argument to `--https-resolver-template`. The default uses the first
+    /// subdomain component.
+    #[clap(long, value_name = "REGEX", default_value = "([a-zA-Z0-9\\-]+)")]
+    https_sni_extract: Regex,
```

Comment:
Is there a reason why this extract Regex is a command line arg? Given how critical it is for resolving connections to environmentd, I would prefer it's a static regex that we have a set of unit tests against. What do you think?

Reply:
Makes sense!

Testing by mocking everything out sounds great, when we get to it, could we also add some unit tests for the supplied regex, and maybe use `proptest`s regex generating ability to test against expected SNIs?

Reply:
The regex just got removed in favor of first subdomain component. So this has been greatly simplified, but will still get a test in the future.